src/Admin/SharedBlockAdmin.php

```diff
@@ -38,9 +38,13 @@ protected function generateBaseRouteName(bool $isChildAdmin = false): string
         return sprintf('%s_%s', parent::generateBaseRouteName($isChildAdmin), 'shared');
     }
 
+    /**
+     * @param DatagridProxyInterface<PageBlockInterface> $query
+     * @return DatagridProxyInterface<PageBlockInterface>&ProxyQueryInterface<PageBlockInterface>
+     */
     protected function configureQuery(ProxyQueryInterface $query): ProxyQueryInterface
     {
-        \assert($query instanceof ProxyQuery);
+        \assert($query instanceof DatagridProxyInterface);
 
         // Filter on blocks without page and parents
         $rootAlias = current($query->getRootAliases());
```

Comment:
all those methods doesn't exists in the ProxyQueryInterface.

But you can solve all the issue by doing

`$query->getQueryBuilder()->getRootAliases()` and so on.